tensorflow/c/kernels.cc

```diff
@@ -92,6 +128,29 @@ void AddTypeConstraint(TF_KernelBuilder* kernel_builder, const char* attr_name,
 }
 #undef CASE
 
+#define CASE(type)                                               \
+  case DataTypeToEnum<type>::value: {                            \
+    kernel_builder->cc_builder->TypeConstraint<type>(attr_name); \
+    break;                                                       \
+  }
+
+void AddTypeConstraint(TF_AsyncKernelBuilder* kernel_builder,
```

Comment:
If we decide to keep separate structs for `TF_KernelBuilder` and `TF_AsyncKernelBuilder`, we can just make `AddTypeContraint` a templated function (with `T* kernel_builder`) to avoid duplicate code.

Reply:
I dont understand, this is a C API, how do we make it as a template function? As I know, template is not supported in C language.

Reply:
Yes, you are right. Done.